File: binance4j-core/src/main/java/com/binance4j/core/dto/TimeInForce.java
package com.binance4j.core.dto;

/**
 * Time in force to indicate how long an order will remain active before it is
 * executed or expires.
 */
public enum TimeInForce {
	/**
	 * GTC (Good-Til-Canceled) orders are effective until they are executed or
	 * canceled.
	 */
	GTC,
	/**
	 * IOC (Immediate or Cancel) orders fills all or part of an order immediately
	 * and cancels the remaining part of the
	 * order.
	 */
	IOC,
	/**
	 * FOK (Fill or Kill) orders fills all in its entirety, otherwise, the entire
	 * order will be cancelled.
	 */
	FOK
}

File: binance4j-core/src/main/kotlin/com/binance4j/core/dto/TimeInForce.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.core.dto

/**
 * Time in force to indicate how long an order will remain active before it is
 * executed or expires.
 */
enum class TimeInForce {
    /**
     * GTC (Good-Til-Canceled) orders are effective until they are executed or
     * canceled.
     */
    GTC,

    /**
     * IOC (Immediate or Cancel) orders fills all or part of an order immediately
     * and cancels the remaining part of the
     * order.
     */
    IOC,

    /**
     * FOK (Fill or Kill) orders fills all in its entirety, otherwise, the entire
     * order will be cancelled.
     */
    FOK
}